AsyncDisplayKit/ASMapNode.mm

```diff
@@ -206,12 +214,32 @@ - (void)takeSnapshot
                     UIGraphicsBeginImageContextWithOptions(image.size, YES, image.scale);
                     [image drawAtPoint:CGPointZero];
                     
-                    // Get a standard annotation view pin. Future implementations should use a custom annotation image property.
-                    MKAnnotationView *pin = [[MKPinAnnotationView alloc] initWithAnnotation:nil reuseIdentifier:@""];
-                    UIImage *pinImage = pin.image;
-                    CGSize pinSize = pin.bounds.size;
+                    MKAnnotationView *pin;
+                    UIImage *pinImage;
+                    CGSize pinSize;
+                    
+                    // Get a standard annotation view pin if there is no custom annotation block.
+                    if (!self.annotationViewInStaticMap) {
+                      pin = [[MKPinAnnotationView alloc] initWithAnnotation:nil reuseIdentifier:@""];
+                      pinImage = pin.image;
+                      pinSize = pin.bounds.size;
+                    }
                     
                     for (id<MKAnnotation> annotation in annotations) {
+                      if (self.annotationViewInStaticMap) {
```

Comment:
Could you please change `self.` to `strongSelf.` to be consistent with the rest of the function!